stan/math/opencl/matrix_cl.hpp

```diff
@@ -211,10 +211,30 @@ class matrix_cl {
     }
     cl::CommandQueue cmdQueue = opencl_context.queue();
     try {
-      opencl_kernels::sub_block(cl::NDRange(nrows, ncols), A.buffer(),
-                                this->buffer(), A_i, A_j, this_i, this_j, nrows,
-                                ncols, A.rows(), A.cols(), this->rows(),
-                                this->cols(), triangular_view);
+      if (triangular_view == TriangularViewCL::Entire) {
+        cl::size_t<3> src_offset;
```

Comment:
I could find a way to initialize this cl::size_t vectors in a shorter manner :/

Reply:
Hah. this is a silly problem with their API it seems. I think the general purpose solution is something like:
```cpp

template <typename Arg, typename... Ts, typename std::enable_if<std::is_integral<Arg>::value>::type * = nullptr>
auto to_size_t(Arg a, Ts... all) {
  constexpr size_t len = sizeof...(Ts)+1;
  int values[len] = {a, all...};
  cl::size_t<len> s;
  for (size_t i = 0; i < len; i++)
    s[i] = values[i];
  return s;
}
```
(we do all the variadic weirdness to [make sure we're getting ints at compile time](https://wiki.sei.cmu.edu/confluence/display/cplusplus/DCL50-CPP.+Do+not+define+a+C-style+variadic+function)). 

this one is a little hackier and if you wanted to pass more than three, you'd have to specify a template parameter. But it's much less confusing imo and is probably fine forever for us:
```cpp
template <int len=3>
cl::size_t<len> to_size_t(std::vector<int> values) {
  assert(values.size() ==len);
  cl::size_t<len> s;
  for (size_t i = 0; i < len; i++)
    s[i] = values[i];
  return s;
}
```

Called like this:
```cpp

int main() {
  cl::size_t<3> s = to_size_t({1, 2, 3});
  return s[0];
}
```

Reply:
Thanks Sean!! I am going to try and incorporate the second one today, I thinks this one is going to be fine forever, for sure :)